src/dodal/devices/electron_analyser/abstract_analyser_io.py

```diff
@@ -1,20 +1,35 @@
 from abc import ABC, abstractmethod
 from typing import TypeVar
 
-from ophyd_async.core import StandardReadable
-from ophyd_async.epics.core import epics_signal_rw
+import numpy as np
+from ophyd_async.core import (
+    Array1D,
+    SignalR,
+    StandardReadable,
+    derived_signal_r,
+    soft_signal_rw,
+)
+from ophyd_async.epics.adcore import ADBaseIO
+from ophyd_async.epics.core import epics_signal_r, epics_signal_rw
 
 from dodal.devices.electron_analyser.abstract_region import EnergyMode
+from dodal.devices.electron_analyser.util import to_binding_energy
 
 
-class AbstractAnalyserDriverIO(ABC, StandardReadable):
+class AbstractAnalyserDriverIO(ABC, StandardReadable, ADBaseIO):
     """
     Generic device to configure electron analyser with new region settings.
     Electron analysers should inherit from this class for further specialisation.
     """
 
     def __init__(self, prefix: str, name: str = "") -> None:
         with self.add_children_as_readables():
+            # Used for setting up region data acquisition. Read per scan
+            self.region_name = soft_signal_rw(str, initial_value="null")
```

Comment:
should: If these are all read per-scan, they should be in a block marked as Config signals, which are only read once per scan:
```
with self.add_children_as_readables(CONFIG_SIGNAL):
    self.region_name = ...
    self.total_time = ...

with self._add_children_as_readables():
    self.image = ...
```

You can see the [types of readable children here](https://github.com/bluesky/ophyd-async/blob/80e70fc31c56c4b198768717644bf6dce0bb152f/src/ophyd_async/core/_readable.py#L17)- if there's a field/some fields you consider the primary important ones that people want to access 90% of the time (like the position of a motor), then you can also make use of HINTED (